### PR TITLE
Implement filesystem operation on spare flash and UNO WiFI OTA

### PR DESCRIPTION
**Pull request by facchinm**

### CODE REVIEW

File: components/mkspiffs/Makefile [hunk 1]
```diff
@@ -0,0 +1,100 @@
+BUILD_CONFIG_NAME ?=
```
Comment (sandeepmistry) [inline components/mkspiffs/Makefile]: Could we make this a submodule to https://github.com/pellepl/spiffs or a fork?

File: components/spiffs/Kconfig [hunk 2]
```diff
@@ -0,0 +1,173 @@
+menu "SPIFFS Configuration"
```
Comment (sandeepmistry) [inline components/spiffs/Kconfig]: Similar to above could this be a submodule?

File: main/CommandHandler.cpp [hunk 3]
```diff
@@ -989,6 +989,200 @@ int setAnalogWrite(const uint8_t command[], uint8_t response[])
...
+  }
+  memcpy(&response[4], &(st.st_size), sizeof(st.st_size));
+
+  response[2] = 1; // number of parameters
+  response[3] = sizeof(st.st_size); // parameter 1 length
+
+  return 10;
+}
+
+int downloadFile(const uint8_t command[], uint8_t response[]) {
```
Comment (sandeepmistry) [inline main/CommandHandler.cpp]: I might be missing a piece in `arduinoOTA` but is this needed? From what I remember the arduinoOTA binary POSTs to a web server ...

Comment (facchinm) [reply main/CommandHandler.cpp]: I added that method to allow a board to perform OTA even if it's behind a NAT (mostly for Create :slightly_smiling_face: ). I used it also for local deployment 
to reduce the amount of code that must be changed to adapt WiFi101OTA for UNO WiFi r2

**Pull request merged**